### PR TITLE
worked on the event app

### PR DESCRIPTION
**Pull request by BEChukwunwike**
I have worked on the event app.

### CODE REVIEW

File: event/models.py [hunk 1]
```diff
@@ -1,3 +1,12 @@
 from django.db import models
 
 # Create your models here.
+class Event(models.Model):
+    title = models.CharField(max_length=200)
+    description = models.TextField()
+    image = models.ImageField(upload_to='event_images/', null=True, blank=True)
+    date = models.DateTimeField()
```
Comment (davidinmichael) [inline event/models.py]: Add a default value to this, so once an event is created, it takes the current date and time of that timezone.

File: event/models.py [hunk 2]
```diff
@@ -1,3 +1,12 @@
...
+    location = models.CharField(max_length=200)
```
Comment (davidinmichael) [inline event/models.py]: What does location mean here, is it part of the data of an event from the UI?

File: event/views.py [hunk 3]
```diff
@@ -1,3 +1,77 @@
...
+from .models import Event
+import json
 
-# Create your views here.
+class EventListView(APIView):
+    """
+    API view to retrieve a list of all events.
+    """
+    def get(self, request):
+        event = Event.objects.all()
```
Comment (davidinmichael) [inline event/views.py]: We shouldn't fetch all events, it should be specific to a user concern.

File: event/views.py [hunk 4]
```diff
@@ -1,3 +1,77 @@
...
+        event_list = list(event.values())
+        return JsonResponse({'event': event_list})
+
+class EventDetailView(APIView):
+    """
+    API view to retrieve details of a specific event.
+    """
+    def get(self, request, pk):
+        event = get_object_or_404(Event, pk=pk)
+        event_data = {
```
Comment (davidinmichael) [inline event/views.py]: Why not return the event object instead of listing it properties?

File: event/views.py [hunk 5]
```diff
@@ -1,3 +1,77 @@
...
+            'date': event.date,
+            'location': event.location,
+        }
+        return JsonResponse({'event': event_data})
+
+class EventCreateView(LoginRequiredMixin, APIView):
+    """
+    API view to create a new event (requires login).
+    """
+    permission_classes = [IsAdminUser]
```
Comment (davidinmichael) [inline event/views.py]: This 'IsAdminUser' is referring to Django Default Admin User, and we won't be using that. There should be a field in the User model that signifies an Admin User, let's use that.

File: event/models.py (hunk 2 above)
Comment (BEChukwunwike) [reply event/models.py]: The event venue/address. I will change it from location to venue_address

File: event/models.py (hunk 1 above)
Comment (BEChukwunwike) [reply event/models.py]: Noted.

File: event/views.py (hunk 3 above)
Comment (BEChukwunwike) [reply event/views.py]: Noted.

**Pull request merged**